src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1074,7 +1074,7 @@ def get_child(self, *seg_type):
                 return seg
         return None
 
-    def get_children(self, *seg_type):
+    def get_children(self, *seg_type) -> list:
```

Comment:
`List["BaseSegment"]` would be more helpful here.

Reply:
changing this actually causes errors in vanilla mypy - i think adding these types will show edge cases in our code that we'll need to address